### PR TITLE
Link together issues by enabling auditor dependencies

### PR DESCRIPTION
**Pull request by zollman**
To help "roll up" issues into larger groupings (e.g. an IAM user
is out-of-compliance if the managed policy attached to that user
is noncompliant), this patch allows auditors to link issues together,
rolling them up into higher technology types.

This commit builds on the dynamic watchers & auditors framework submitted in https://github.com/Netflix/security_monkey/pull/405 , as well as the changes built into https://github.com/Netflix/security_monkey/pull/406 .

### CODE REVIEW

**Comment by aebie** [comment]
@monkeysecurity I was pulling these PR branches plus the 0.6.0 release back into out local repo and notices some tricky merge issues with the custom accounts branch. It looks like you will be pulling in this branch first? If so, I can go ahead and resolve the merge conflicts in custom accounts.

**Comment by monkeysecurity** [comment]
Hey @aebie,

Yeah, I was going to merge this one #407 before #408.

These two will be in a separate release (0.7.5).  I've been spending this week looking for any bugs in  the 0.7.0 release before I push that out.

**Comment by aebie** [comment]
@monkeysecurity I will work on an update to merge 407 into 408 then. We do have one outstanding bug fix for 407 that is in the review process, but has to do with auditors that have more than one support watcher so it will not manifest with the auditors currently submitted so probably is not critical to get in right away.

**Comment by monkeysecurity** [comment]
Applying "Needs Work" label as I believe Curtis is working on updating this PR.  (See gitter.im chat from Oct 13 15:55)

**Comment by cstewart87** [comment]
UDPATE:  still working on these updates.  Hope to post w/in 1 week

**Comment by cstewart87** [comment]
Build will probably fail until https://github.com/Netflix/security_monkey/pull/438 is merged

**Comment by monkeysecurity** [comment]
I merged 438 and instructed travis to rebuild.  Build is now working.

I think I'm still waiting for new commits.  Is that correct?

**Comment by cstewart87** [comment]
@monkeysecurity - we just pushed some updates out today.  Should have all of the commits needed for review 👍

**Comment by cstewart87** [comment]
@monkeysecurity - not sure why the dates are funky, prob something to do with cherry-picking, but the 2 latest commits listed under "Oct 18" section (https://github.com/Netflix/security_monkey/pull/407/commits) are what was pushed today

File: security_monkey/watchers/config_recorder.py [hunk 1]
```diff
@@ -20,20 +20,32 @@
...
 
 class ConfigRecorder(Watcher):
     index = 'configrecorder'
     i_am_singular = 'Config Recorder'
     i_am_plural = 'Config Recorders'
 
+    # TODO: Replace hardcoded region with `get_available_regions` after next boto3 release
+    # Specific PR here: https://github.com/boto/boto3/pull/531/files
+    # AWS Config currently only supports the following regions
+    regions = ["us-east-1", "us-west-1", "us-west-2", "eu-west-1", "eu-central-1",
```
Comment (monkeysecurity) [inline security_monkey/watchers/config_recorder.py]: That boto3 pr was merged on Mar3.  Is the hardcoded region list still necessary?

Comment (aebie) [inline security_monkey/watchers/config_recorder.py]: @monkeysecurity They are not. Let me look into why they are still there.

Comment (aebie) [inline security_monkey/watchers/config_recorder.py]: @monkeysecurity we have the correct code internally and it looks like it is correct in you master. We will push a fix for this branch.

**Pull request merged**